compiler/cli/cli-common/src/org/jetbrains/kotlin/utils/parametersMap.kt

```diff
@@ -218,10 +219,31 @@ private class AnyArgsConverter : ArgsConverter<Any> {
                 else -> null
             }
 
-        if (value::class.isSubclassOf(parameter.type.jvmErasure)) return ArgsConverter.Result.Success(value)
+        fun evaluateValue(arg: Any): Any? {
+            if (arg::class.isSubclassOf(parameter.type.jvmErasure)) return arg
+            return convertPrimitivesArray(parameter.type, arg)
+        }
+
+        evaluateValue(value)?.let { return ArgsConverter.Result.Success(it) }
+
+        // Handle special scenario where [arg::class] is Array<Any> or Array<*> but
+        // but it's values all contain the proper values
+        val arrayComponentType = parameter
+            .type
+            .classifier
+            .safeAs<KClass<*>>()
+            ?.takeIf { it.java.isArray }
+            ?.java
+            ?.componentType
+            ?.kotlin
```

Comment:
Please don't use such a chaining style, it hinder the readability. Group calls by some property, e.g. `parameter.type.classifier` can be placed on one line.

Reply:
✅